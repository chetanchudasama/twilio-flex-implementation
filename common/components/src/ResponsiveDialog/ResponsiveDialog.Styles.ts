import Dialog from "@material-ui/core/Dialog";
import styled from "react-emotion";
import { black, borderStyle } from "../shared/AppTheme";

export const ResponsiveDialogStyles = styled(Dialog)({
  fontFamily: "Open Sans !important",
  ".dialog-title": {
    borderBottom: borderStyle,
  },
  ".dialog-title > h6": {
    fontSize: "22px",
    fontWeight: 600,
    textAlign: "left",
    color: black,
  },
  ".confirmation-message": {
    margin: "10px",
  },
  ".disabled-btn": {
    pointerEvents: "none",
    cursor: "default",
    opacity: "0.4",
  },
  ".close-icon": {
    position: "absolute",
    right: 8,
    top: 5,
    padding: 4,
  },
  ".delete-btn": {
    position: "absolute",
    left: "12px",
  },
  ".delete-btn-icon": {
    margin: "-3px 5px 0 0",
    fontSize: 20,
  },
});
