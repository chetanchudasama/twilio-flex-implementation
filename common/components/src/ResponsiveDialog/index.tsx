import Button from "@material-ui/core/Button";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import React from "react";
import styled from "react-emotion";
import { IconButton, Icon } from "@material-ui/core";
import CloseIcon from "@material-ui/icons/Close";
import { ResponsiveDialogStyles } from "./ResponsiveDialog.Styles";
import { white, secondary, deleteButtonColor } from "../shared/AppTheme";

export interface ResponsiveDialogProps {
  open: boolean;
  title?: string;
  okText?: string;
  cancelText?: string;
  deleteText?: string;
  maxWidth?: "xs" | "sm" | "md" | "lg" | "xl" | false;
  showActionButtons?: boolean;
  onConfirm: () => void;
  onCancel: () => void;
  isConfirmationDialog?: boolean;
  disableSaveButton?: boolean;
  titleCloseIcon?: boolean;
  showDeleteButton?: boolean;
  onDelete?: () => void;
}

export const ResponsiveDialog: React.FC<ResponsiveDialogProps> = ({
  maxWidth,
  open,
  onCancel,
  onConfirm,
  title,
  titleCloseIcon,
  isConfirmationDialog,
  showActionButtons,
  cancelText,
  okText,
  deleteText,
  children,
  showDeleteButton,
  disableSaveButton,
  onDelete,
}) => {
  const StyledSaveButton = styled(Button)({
    backgroundColor: `${secondary}!important`,
    color: `${white}!important`,
  });

  const StyledCancelButton = styled(Button)({
    backgroundColor: `${white}!important`,
    color: `${secondary}!important`,
  });

  const StyledDeleteButton = styled(Button)({
    backgroundColor: `${white}!important`,
    color: `${deleteButtonColor}!important`,
  });

  return (
    <ResponsiveDialogStyles
      fullWidth
      maxWidth={maxWidth || "sm"}
      open={open}
      onClose={onCancel}
      aria-labelledby="max-width-dialog-title"
      disableBackdropClick
      disableEscapeKeyDown
      scroll="paper"
      disableEnforceFocus
    >
      {title || titleCloseIcon ? (
        <DialogTitle
          className={!titleCloseIcon ? "dialog-title" : ""}
          id="max-width-dialog-title"
        >
          {title || ""}
          {titleCloseIcon ? (
            <IconButton
              aria-label="Close"
              className="close-icon"
              onClick={onCancel}
            >
              <CloseIcon />
            </IconButton>
          ) : null}
        </DialogTitle>
      ) : null}
      <DialogContent
        className={isConfirmationDialog ? "confirmation-message" : ""}
      >
        {children}
      </DialogContent>
      {showActionButtons && (
        <DialogActions>
          {showDeleteButton && (
            <StyledDeleteButton
              variant="text"
              onClick={onDelete}
              type="button"
              className="delete-btn"
            >
              <Icon className="delete-btn-icon">delete</Icon>
              {deleteText || "Delete"}
            </StyledDeleteButton>
          )}
          <StyledCancelButton variant="text" onClick={onCancel} type="button">
            {cancelText || "Close"}
          </StyledCancelButton>
          <StyledSaveButton
            variant="contained"
            type="button"
            onClick={onConfirm}
            className={disableSaveButton ? "disabled-btn" : ""}
          >
            {okText || "Save"}
          </StyledSaveButton>
        </DialogActions>
      )}
    </ResponsiveDialogStyles>
  );
};
