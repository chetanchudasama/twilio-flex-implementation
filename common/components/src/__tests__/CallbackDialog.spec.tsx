/* eslint-disable react/jsx-props-no-spreading */
import { create, act } from "react-test-renderer";
import React from "react";
import { shallow } from "enzyme";
import {
  CallbackDialog,
  CallbackDialogProps,
} from "../CallbackDialog/CallbackDialog";
import { ResponsiveDialog } from "../ResponsiveDialog";

const props: CallbackDialogProps = {
  open: true,
  setCallbackDetail: jest.fn(),
  handleCloseDialog: jest.fn(),
};

jest.mock("../ResponsiveDialog/index", () => {
  return {
    ResponsiveDialog: jest.fn(({ children }) => children),
  };
});

jest.mock("../MuiDatePicker/MuiDatePicker", () => {
  return {
    MuiDatePicker: () => {
      return <div>Date Picker</div>;
    },
  };
});

describe("CallbackDialog", () => {
  it("renders correctly", () => {
    const wrapper = create(<CallbackDialog {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should not render Reason text-files, when value of noteOptional prop is true", () => {
    props.noteOptional = true;
    const wrapper = shallow(<CallbackDialog {...props} />);

    expect(wrapper.find(".reason-field").exists()).toBeFalsy();
  });

  it("should call handleCloseDialog method, when cancel button is clicked from callback dialog", () => {
    props.noteOptional = false;
    const wrapper = shallow(<CallbackDialog {...props} />);

    act(() => {
      wrapper.find(ResponsiveDialog).last().props().onCancel();
    });

    expect(props.handleCloseDialog).toHaveBeenCalled();
  });

  it("when delete button is clicked, should open confirmation dialog", () => {
    const wrapper = shallow(<CallbackDialog {...props} />);

    expect(wrapper.find(ResponsiveDialog)).toHaveLength(1);

    act(() => {
      wrapper.children().last().props().onDelete();
    });
    wrapper.update();

    expect(wrapper.find(ResponsiveDialog)).toHaveLength(2);
    expect(wrapper.find(".confirmation-message").exists()).toBeTruthy();
  });

  it("should close confirmation dialog, when No button is clicked from confirmation dialog", () => {
    const wrapper = shallow(<CallbackDialog {...props} />);

    act(() => {
      wrapper.children().last().props().onDelete();
    });
    wrapper.update();

    expect(wrapper.find(".confirmation-message").exists()).toBeTruthy();

    act(() => {
      wrapper.find(ResponsiveDialog).last().props().onCancel();
    });

    expect(wrapper.find(".confirmation-message").exists()).toBeFalsy();
  });
});
