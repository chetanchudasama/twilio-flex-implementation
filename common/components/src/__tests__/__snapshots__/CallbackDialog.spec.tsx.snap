// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallbackDialog renders correctly 1`] = `
<div
  className="css-1k3gdne"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div>
      Date Picker
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 reason-field"
  >
    <div
      className="MuiFormControl-root-98 MuiFormControl-fullWidth-101"
    >
      <label
        className="MuiFormLabel-root-113 MuiFormLabel-required-118 MuiInputLabel-required-106 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110"
        data-shrink={false}
      >
        Reason for call
        <span
          className="MuiFormLabel-asterisk-119"
        >
          â€‰*
        </span>
      </label>
      <div
        className="MuiInputBase-root-133 MuiInput-root-120 MuiInput-underline-124 MuiInputBase-fullWidth-142 MuiInput-fullWidth-127 MuiInputBase-formControl-134 MuiInput-formControl-121 MuiInputBase-multiline-141 MuiInput-multiline-126"
        onClick={[Function]}
      >
        <textarea
          aria-invalid={false}
          className="MuiInputBase-input-143 MuiInput-input-128 MuiInputBase-inputMultiline-145 MuiInput-inputMultiline-130"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          rows={3}
          value=""
        />
      </div>
      
    </div>
  </div>
</div>
`;
